File: javascript/sorting/p040_bubbleSort.js
/*
    CLRS Chapter 2 Problem 2-2, p. 40
*/

const bubbleSort = (A) => {
    for (let i = 0; i < A.length; i++) {
        for (let j = A.length - 1; j >= i + 1; j--) {
            if (A[j] < A[j - 1]) {
                [A[j], A[j - 1]] = [A[j - 1], A[j]];
            }
        }
    }
}

module.exports = {
    bubbleSort
}

File: python/sorting/p040_bubble_sort.py
'''
    CLRS Chapter 2 Problem 2-2, p. 40
'''

def bubble_sort(A):
    for i in range(0, len(A)):
        for j in range(len(A)-1, i, -1):
            if A[j] < A[j - 1]:
                A[j], A[j - 1] = A[j - 1], A[j]
